### PR TITLE
feat: function to look up synonymous HDF5 node names

### LINKED ISSUE #3: Handling of peregrine_hdf5 synonymous process parameter names
The Peregrine HDF5 metadata is loaded into Peregrine via a tabulated file that can have variable column names. There is no data cleaning process that enforces specific column names, so there is the possibility of synonymous process parameter names, e.g., "scan_speed" and "velocity" for describing the speed at which the laser moves during scanning.

The column names that are saved in the HDF5 file that Peregrine exports are derived from a manually written tabulated file. The following operations are performed to convert the column names in the tabulated file to the HDF5 node labels using the following logic:

- The column names (`col`) are split into name and unit by `col_name = col.split(" (")[0]`
- The columns names are then converted by `col_name = col_name.lower().replace(" ", "_")`

These column names are not necessarily standardized. While that may change in the future, there is not currently a timeline for when.

### PR DESCRIPTION
**Pull request by gknapp1**
Adds core functionality for looking up values from a dictionary-like object by passing a list of synonymous keys. This is implemented in the `peregrine` and `peregrine_hdf5` databases to handle potentially varying node names for process parameters, as described in #3.

Closes #3.

### CODE REVIEW

File: myna/core/utils/nested_dict_tools.py [hunk 1]
```diff
@@ -21,3 +21,20 @@ def nested_get(dict, keys):
     for key in keys[:-1]:
         dict = dict.setdefault(key, {})
     return dict[keys[-1]]
+
+
+def get_synonymous_node(dict_obj, synonym_list):
```
Comment (streeve) [inline myna/core/utils/nested_dict_tools.py:26]: What does `node` refer to?

File: myna/database/peregrine.py [hunk 2]
```diff
@@ -49,11 +49,9 @@ def load(self, metadata_type, part=None, layer=None):
         if metadata_type == metadata.LaserPower:
             datafile = os.path.join(self.path, "simulation", part, "part.npz")
             with np.load(datafile, allow_pickle=True) as data:
-                index = [
-                    ind
-                    for ind, x in enumerate(data["parameter_names"])
-                    if x == "Power (W)"
-                ][0]
+                synonyms = ["Laser Beam Power (W)", "Power (W)"]
+                matches = [syn in data["parameter_names"] for syn in synonyms]
```
Comment (streeve) [inline myna/database/peregrine.py:53]: Could use the same util function?

**Review by streeve — CHANGES_REQUESTED** [review]

**Comment by gknapp1** [comment]
In addition to figuring out the current CI error, I am going to create a minimal HDF5 resource based on the Spatial Variation Baseline database and a test for checking `PeregrineHDF5` functionality.

File: myna/database/peregrine.py (hunk 2 above)
Comment (gknapp1) [reply myna/database/peregrine.py]: Looks good to me! Cleaner and more consistent than what was there.

**Review by gknapp1 — COMMENTED** [review]
The requested changes look good to me. All the features that I was looking to add in this PR are implemented now, so should be ready to merge.

**Review by streeve — APPROVED** [review]

**Pull request merged**